python_modules/dagster/dagster/_core/definitions/asset_condition.py

```diff
@@ -18,17 +21,35 @@
 )
 from dagster._core.definitions.events import AssetKey
 from dagster._core.definitions.metadata import MetadataMapping, MetadataValue
-from dagster._serdes.serdes import whitelist_for_serdes
+from dagster._core.definitions.partition import AllPartitionsSubset
+from dagster._serdes.serdes import (
+    FieldSerializer,
+    UnpackContext,
+    WhitelistMap,
+    pack_value,
+    unpack_value,
+    whitelist_for_serdes,
+)
+from dagster._utils.caching_instance_queryer import CachingInstanceQueryer
 
 from .asset_condition_evaluation_context import (
     AssetConditionEvaluationContext,
 )
 from .asset_subset import AssetSubset
 
 if TYPE_CHECKING:
+    from dagster._core.definitions.asset_graph import AssetGraph
+
     from .auto_materialize_rule import AutoMaterializeRule
 
 
+@whitelist_for_serdes
+class HistoricalAllPartitionsSubset(NamedTuple):
```

Comment:
The naming of this class makes it seem like it should be a subclass of `PartitionsSubset`. Maybe tack `Sentinel` on the end?

Reply:
done!